Data/HashMap/Array.hs

```diff
@@ -321,7 +321,7 @@ updateWith' ary idx f = update ary idx $! f (index ary idx)
 -- inserting it into the array.
 updateWithInternal' :: Array e -> Int -> (e -> (Int, e)) -> RunResA e
 updateWithInternal' ary idx f =
-    let (!sz, !e) = f (index ary idx)
+    let !(!sz, !e) = f (index ary idx)
```

Comment:
Sorry, I didn't mean to confuse matters. You need the outer bang, yes, but you should probably also get rid of the tuple in favor of a custom strict type. I generally push tuples to the edges, where users lie. They aren't usually what we want internally here, IMO. Their laziness is useful for certain purposes, but when we don't need it we're better off without it.